managed/models/agentversion.go

```diff
@@ -37,18 +36,22 @@ func (e *AgentNotSupportedError) Error() string {
 	return fmt.Sprintf("'%s' functionality is not supported by pmm-agent %q version %q. Required minimum version is %q", e.Functionality,
 		e.AgentID, e.AgentVersion, e.MinAgentVersion)
 }
+func (e *AgentNotSupportedError) Is(err error) bool {
+	_, ok := err.(*AgentNotSupportedError)
+	return ok
+}
```

Comment:
Why it's needed? It's analog of `errors.Is()`, no?

Reply:
errors.Is didn't work correctly and actually calls this method to compare, so I had to implement it this way.

Reply:
https://github.com/percona/pmm/pull/2625/files#diff-8792e067d067e501754c905b86a46d9101c2861f211cc265cdc5fd395a2e4bc4R81 here is where errors.Is is called

Reply:
It seems that we need `As` there instead of `Is`. We need to check error type, but not exact copy.

Reply:
As didn't work as well

Reply:
Why? What's so special about this case?

Reply:
I have no idea

Reply:
the reason was because `IsAgentSupported` was returning a pointer and that's why it didn't work.